### PR TITLE
Improve and update ResourceQuota for 1.2

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Update the example for quota to reflect latest state of system.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
@pmorie @ncdc - can either of you provide the tech LGTM ?

@bgrant0607 - fyi, as this documents an example that uses scopes on quota and updates stuff to use deployments per the latest code.

**Comment by derekwaynecarr** [comment]
related: https://github.com/kubernetes/kubernetes.github.io/pull/620

**Comment by pmorie** [comment]
@derekwaynecarr rebase needed

**Comment by bgrant0607** [comment]
@derekwaynecarr The master branch is currently live on kubernetes.io. Could you please make 1.3-specific changes in the release-1.3 branch?
https://github.com/kubernetes/kubernetes.github.io/tree/release-1.3

cc @johndmulhausen

**Comment by derekwaynecarr** [comment]
Ok, the 1.3 specific changes are just services.nodeports and
services.loadbalancers.

Will update tomorrow.

**Comment by derekwaynecarr** [comment]
@ncdc - this PR now includes the updates for both documentation and examples in 1.2 in a single PR to make it easier to manage.  PTAL, sorry for confusion across PRs but filenames got changed (index ->walkthrough) and it made collapsing things into a single PR much easier.

File: docs/admin/resourcequota/index.md [hunk 1]
```diff
@@ -4,29 +4,26 @@
...
-  [admission controller](/docs/admin/admission-controllers)) before the quota is checked to avoid this problem.
+- The administrator creates one or more Resource Quota objects for each namespace.
+- Users create resources (pods, services, etc.) in the namespace, and the quota system
+  tracks usage to ensure it does not exceed hard resource limits defined in a Resource Quota.
+- If creating or updating a resource violates a quota constraint, the request will fail with HTTP
+  status code `403 FORBIDDEN` with a message explaining the constraint that would have been violated.
+- If quota is enabled in a namespace for compute resources like `cpu` and `memory`, users must specify
+  requests or limits for those values; otherwise, the quota system may reject pod creation.  Hint: Use
+  the LimitRange admission controller to force defaults for pods that make no compute resource requirements.
+  See the [walkthrough](/docs/admin/resourcequota/walkthrough.md) for an example to avoid this problem.
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/index.md]: Correct URL is /docs/admin/resourcequota/walkthrough/

File: docs/admin/resourcequota/index.md [hunk 2]
```diff
@@ -57,11 +54,12 @@ in a namespace can be limited.  The following compute resource types are support
 
 | ResourceName | Description |
 | ------------ | ----------- |
-| cpu | Total cpu requests of containers |
-| memory | Total memory requests of containers
-
-For example, `cpu` quota sums up the `resources.requests.cpu` fields of every
-container of every pod in the namespace, and enforces a maximum on that sum.
+| cpu | Across all pods in a non-terminal state, the sum of CPU requests cannot exceed this value. |
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/index.md]: Let's have all the Resource Names be surrounded in `backticks` (and `ResourceName` itself if that is the name of an object)

File: docs/admin/resourcequota/index.md [hunk 3]
```diff
@@ -84,45 +83,105 @@ You might want to set a pods quota on a namespace
...
+Resources specified on the quota outside of the allowed set results in a validation error.
+
+| Scope | Description |
+| ----- | ----------- |
+| Terminating | Match pods where `spec.activeDeadlineSeconds >= 0` |
+| NotTerminating | Match pods where `spec.activeDeadlineSeconds is nil` |
+| BestEffort | Match pods that have best effort quality of service. |
+| NotBestEffort | Match pods that do not have best effort quality of service. |
+
+The `BestEffort` scope restricts a quota to tracking the following resources:
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/index.md]: s/the following resource: *pods/pods

File: docs/admin/resourcequota/index.md [hunk 4]
```diff
@@ -84,45 +83,105 @@ You might want to set a pods quota on a namespace
...
+* pods
+
+The `Terminating`, `NotTerminating`, and `NotBestEffort` scopes restrict a quota to tracking the following resources:
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/index.md]: Blank line between text and list, please

File: docs/admin/resourcequota/walkthrough.md [hunk 5]
```diff
@@ -1,164 +1,350 @@
...
 ---
 
-This example demonstrates how [resource quota](/docs/admin/admission-controllers/#resourcequota) and
-[limitsranger](/docs/admin/admission-controllers/#limitranger) can be applied to a Kubernetes namespace.
-See [ResourceQuota design doc](https://github.com/kubernetes/kubernetes/blob/{{page.githubbranch}}/docs/design/admission_control_resource_quota.md) for more information.
+This example demonstrates a typical setup to control for resource usage in a namespace.
+
+It demonstrates using the following resources:
+* [Namespace](/docs/admin/namespaces)
+* [Resource Quota](/docs/admin/admission-controllers/#resourcequota)
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/walkthrough.md]: Wouldn't a better link be /docs/admin/resourcequota/ ?

File: docs/admin/resourcequota/walkthrough.md [hunk 6]
```diff
@@ -1,164 +1,350 @@
...
+* [Limit Range](/docs/admin/admission-controllers/#limitranger)
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/walkthrough.md]: Wouldn't a better link be /docs/admin/limitrange/ ?

File: docs/admin/resourcequota/walkthrough.md [hunk 7]
```diff
@@ -1,164 +1,350 @@
...
 
 This example assumes you have a functional Kubernetes setup.
 
+## Scenario
+
+The cluster-admin is operating a cluster on behalf of a user population and the cluster-admin
+wants to control the amount of resource that can be consumed in a particular namespace to promote
+fair sharing of the cluster and control cost.
+
+The cluster-admin has the following goals:
```
Comment (johndmulhausen) [inline docs/admin/resourcequota/walkthrough.md]: Blank line between text and list please

**Comment by johndmulhausen** [comment]
@derekwaynecarr , thanks so much for splitting the 1.3 changes out. Some things to address but they're mostly formatting stuff.

**Comment by derekwaynecarr** [comment]
@ncdc @johndmulhausen - updates made

**Comment by derekwaynecarr** [comment]
@johndmulhausen - see https://github.com/kubernetes/kubernetes.github.io/issues/719 which causes similar issues with this pr when rendered in final page.

**Comment by derekwaynecarr** [comment]
self applying tech lgtm as its the same content from the 1.3 pr just reduced.

**Comment by johndmulhausen** [comment]
Thanks for addressing the comments, @derekwaynecarr ! Merging.

**Pull request merged**